### PR TITLE
WIP: Bincurat

### PR DESCRIPTION
**Pull request by radare**
Random related comments:

* Should be r_bin_file_current()... but.. we really want to have that? i mean, a default binfile? and we should use descriptors, not pointers
* same happens with R_IO which assumes there's a current "filedescriptor".. this is in siol-speak used only when working in non-va mode

This function spots a bunch of bad design decisions in RBin and its not used anywhere yet, lets keep it as is for now.

### CODE REVIEW

**Comment by radare** [comment]
this function spots a bunch of bad design decisions in RBin and its not used anywhere yet, lets keep it as is for now

File: libr/bin/bfile.c [hunk 1]
```diff
@@ -904,3 +906,24 @@ R_API int r_bin_file_hash(RBin *bin, ut64 limit, const char *file) {
...
+	RListIter *iter;
+	RBinFile *obf = bin->cur; // XXX this must die
+	r_list_foreach (bin->binfiles, iter, bf) {
+		bin->cur = bf;
+		RList *sections = r_bin_get_sections (bin);
+		RBinSection *s;
+		r_list_foreach (sections, iter, s) {
+			ut64 from = s->vaddr;
+			ut64 to = from + s->vsize;
+			if (R_BETWEEN (from, addr, to)) {
```
Comment (ret2libc) [inline libr/bin/bfile.c]: the bad about R_BETWEEN is that it includes both the start and the end, so i think you should use `to - 1`

**Comment by radare** [comment]
this is important to support multidex

**Comment by radare** [comment]
rebased

**Comment by radare** [comment]
Its now merged an already working implementation. `e bin.at=true`

**Pull request closed without merging**